src/api/types.h

```diff
@@ -23,5 +28,74 @@ typedef struct CGameSettings {
   const char* seat_controllers[4];
   int num_controllers = 0;
 };
+
+typedef enum CMeldType {
+  kMeldChi = 0,
+  kMeldPon = 1,
+  kMeldKan = 2,
+  kMeldConcealedKan = 3,
+  kMeldNone = 4 // Default value
+} CMeldType;
+
+// C API version of StateFunctionType enum
+typedef enum CStateFunctionType {
+  kError = 0,
+  kGameStart = 1,
+  kRoundStart = 2,
+  kDraw = 3,
+  kPlayerHand = 4,
+  kPon = 5,
+  kChi = 6,
+  kKan = 7,
+  kConcealedKan = 8,
+  kConvertedKan = 9,
+  kKanDiscard = 10,
+  kReplacement = 11,
+  kRiichi = 12,
+  kDiscard = 13,
+  kExhaust = 14,
+  kRon = 15,
+  kTsumo = 16,
+  kRoundEnd = 17,
+  kGameEnd = 18
+} CStateFunctionType;
+
+typedef struct CMeld {
+  CMeldType type;
+  CPiece start;
+};
+
+typedef struct CHand {
+  CPiece live[kMaxLiveHandSize];           // Maximum live pieces
```

Comment:
Oh I see what you mean by live pieces. I think probably better to just have this as pieces. I think the distinction of live might be confusing.

Reply:
lmao this is my own terminology you're reusing here.
ok in that case can we just rename this to livePieces.

Reply:
yes lmao sounds good